### PR TITLE
Cover community sharding with tests.

### LINKED ISSUE #4262: Cover community sharding with tests
# Problem

We have recently implemented community sharding. The following scenarios must be covered with tests:

1) Members should be able to receive messages in a community with sharding enabled.
2) Members should continue to receive messages in a community if sharding is disabled after it was previously enabled.
3) Members should be able to receive messages in a community where the sharding configuration has been edited.

### PR DESCRIPTION
**Pull request by osmaczko**
closes: #4262

Review commit by commit, the most interesting file is: `protocol/messenger_communities_sharding_test.go`

### CODE REVIEW

File: protocol/communities_messenger_helpers_test.go [hunk 1]
```diff
@@ -149,58 +148,45 @@ func (c *CollectiblesServiceMock) DeploymentSignatureDigest(chainID uint64, addr
...
-	return messenger
+
+	return nil
 }
 
-func newCommunitiesTestMessenger(shh types.Waku, privateKey *ecdsa.PrivateKey, logger *zap.Logger, accountsManager account.Manager,
-	tokenManager communities.TokenManager, collectiblesService communitytokens.ServiceInterface) (*Messenger, error) {
-	config := params.NodeConfig{
+func defaultTestCommunitiesMessengerNodeConfig() *params.NodeConfig {
+	// config.WakuV2Config.UseShardAsDefaultTopic = true
```
Comment (igor-sirotin) [inline protocol/communities_messenger_helpers_test.go]: leftover?

File: protocol/communities_messenger_helpers_test.go [hunk 2]
```diff
@@ -221,29 +207,56 @@ func newCommunitiesTestMessenger(shh types.Waku, privateKey *ecdsa.PrivateKey, l
...
+}
+
+func newTestCommunitiesMessenger(s *suite.Suite, waku types.Waku, config testCommunitiesMessengerConfig) *Messenger {
+	err := config.complete()
+	s.Require().NoError(err)
+
+	accountsManagerMock := &AccountManagerMock{}
+	accountsManagerMock.AccountsMap = make(map[string]string)
+	for _, walletAddress := range config.walletAddresses {
+		accountsManagerMock.AccountsMap[walletAddress] = types.EncodeHex(crypto.Keccak256([]byte(config.password)))
```
Comment (igor-sirotin) [inline protocol/communities_messenger_helpers_test.go:218]: `types.EncodeHex(crypto.Keccak256([]byte(config.password)))` can be extracted to a variable

File: protocol/communities_messenger_helpers_test.go [hunk 3]
```diff
@@ -221,29 +207,56 @@ func newCommunitiesTestMessenger(shh types.Waku, privateKey *ecdsa.PrivateKey, l
 		LinkPreviewRequestEnabled: true,
 		SendStatusUpdates:         true,
 		WalletRootAddress:         types.HexToAddress("0x1122334455667788990011223344556677889900")}
+}
+
+func newTestCommunitiesMessenger(s *suite.Suite, waku types.Waku, config testCommunitiesMessengerConfig) *Messenger {
```
Comment (igor-sirotin) [inline protocol/communities_messenger_helpers_test.go:211]: should we maybe add `extraOptions []Option` here, like in `newTestMessenger`?
I will definitely use it (have code not commited yet), but I could also add it myself ofc.

File: protocol/communities_events_utils_test.go [hunk 4]
```diff
@@ -1115,14 +1115,6 @@ func testAcceptMemberRequestToJoin(base CommunityEventsTestsInterface, community
...
 	s.Require().Equal(acceptedRequestsPending[0].PublicKey, common.PubkeyToHex(&user.identity.PublicKey))
 
-	// user should not receive community admin event without being a member yet
-	_, err = WaitOnMessengerResponse(
-		user,
-		func(r *MessengerResponse) bool { return len(r.Communities()) > 0 },
-		"user did not receive community request to join response",
-	)
-	s.Require().Error(err)
-
```
Comment (igor-sirotin) [inline protocol/communities_events_utils_test.go:1125]: Why is this?

File: protocol/messenger_communities_sharding_test.go [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+		s.Require().NoError(err)
+
+		message := buildTestMessage(*chat)
+		_, err = s.owner.SendChatMessage(context.Background(), message)
+		s.Require().NoError(err)
+
+		_, err = WaitOnMessengerResponse(s.alice, func(mr *MessengerResponse) bool {
+			return len(mr.messages) > 0 && mr.Messages()[0].ID == message.ID
+		}, "message not received")
+		s.Require().NoError(err)
```
Comment (igor-sirotin) [inline protocol/messenger_communities_sharding_test.go]: This code is repeated 3 times with a few changes, could we extract it to a function with parameters?

**Review by igor-sirotin — APPROVED** [review]
Great! 🔥

File: protocol/communities_events_utils_test.go (hunk 4 above)
Comment (osmaczko) [reply protocol/communities_events_utils_test.go:1125]: This check didn't make any logical sense; for sure, it didn't check 'user should not receive a community admin event without being a member yet.' Removing it keeps tests green.

File: protocol/communities_messenger_helpers_test.go (hunk 3 above)
Comment (osmaczko) [reply protocol/communities_messenger_helpers_test.go:211]: Please add it yourself; I am not a fan of adding code that is not used anywhere. Thanks :pray:

**Pull request merged**